### DISCUSSION TITLE
Database Check Constraints

### DISCUSSION THREAD

**Discussion by ionsource** [oqtane/oqtane.framework#1626] [Q&A]
Is it possible to add a database Check Constraint via the a Oqtane module migration (e.g. check that a start time is before a stop time)? I believe support for Check Constraints became available in the EF Core 3 release.

**Comment by sbwalker** [comment]
I have never tried to use Check Constraints in Oqtane so I cannot comment on whether it would work or not. However you should keep in mind that Oqtane uses an abstraction over EF Core in order to support multi-tenancy and modularity so you may not have the full scope of EF Core available currently - it may require a core enhancement. That being said, I would personally avoid relying on business logic at the database layer as it often creates dependencies on the capabilities of a specific database. I would implement these validation rules in code.

**Comment by ionsource** [comment]
I got this working using the following method in my migration:
            migrationBuilder.Sql($@"
                ALTER TABLE table ADD CONSTRAINT CK_table_JSON check(data IS JSON);
                ");
In general I agree with avoiding business logic in the database, but in this case I have a CLOB that stores JSON data, and need the database check for the direct query of data inside the JSON CLOB.

**Reply by W6HBR** [reply]
@ionsource I just wanted to say that this little nugget helped me immensely!  I was struggling with how to add named constraints as well as how to add a constraint that sets a default using newid().  Your solution above was exactly what I needed. Thank you.
